pkg/autodiscovery/listeners/kube_endpoints.go

```diff
@@ -307,6 +315,22 @@ func (l *KubeEndpointsListener) createService(kep *v1.Endpoints, checkServiceAnn
 
 	eps := processEndpoints(kep, tags)
 
+	for i := 0; i < len(eps); i++ {
+		eps[i].metricsExcluded = l.containerFilters.IsExcluded(
+			containers.MetricsFilter,
+			kep.GetAnnotations(),
+			kep.Name,
+			"",
+			kep.Namespace,
+		) || l.containerFilters.IsExcluded(
+			containers.GlobalFilter,
+			kep.GetAnnotations(),
+			kep.Name,
+			"",
+			kep.Namespace,
+		)
+	}
+
```

Comment:
maybe it can be good to check before that `l.containerFilters` is not `nil` for running the for loop.

Reply:
i've changed it to still loop but to default to `false` for both `metricsExcluded` and `globalExcluded` if `l.containerFilters` is `nil`
question: since `l.containerFilters` is typed to `containerFilters`, https://github.com/DataDog/datadog-agent/blob/9117adcc4aacac24a139336e0f3610e352182ef7/pkg/autodiscovery/listeners/common.go#L30-L34 would it still be possible for this branch to run?